src/ripple/rpc/handlers/Feature1.cpp

```diff
@@ -67,6 +72,10 @@ doFeature(RPC::JsonContext& context)
     if (!feature && !feature.parseHex(context.params[jss::feature].asString()))
         return rpcError(rpcBAD_FEATURE);
 
+    if (!isAdmin && context.params.isMember(jss::vetoed))
+    {
+        return rpcError(rpcNO_PERMISSION);
+    }
     if (context.params.isMember(jss::vetoed))
     {
         if (context.params[jss::vetoed].asBool())
```

Comment:
You could make this a lot safer and easier to reason about by simply moving the `isAdmin` check inside the voting block.
```
    if (context.params.isMember(jss::vetoed))
    {
        if (!isAdmin)
            return rpcError(rpcNO_PERMISSION);

        if (context.params[jss::vetoed].asBool())
            table.veto(feature);
        else
            table.unVeto(feature);
    }
```

So now the "voting" code is all in one block, and the logic is like
1. Am I trying to vote?
   1. Am I *allowed* to vote?
      1. No? Fail!
      2. Yes? Vote!